hostpath-provisioner.go

```diff
@@ -94,12 +103,24 @@ func isCorrectNode(annotations map[string]string, nodeName string) bool {
 	return false
 }
 
+func (p *hostPathProvisioner) ShouldProvision(pvc *v1.PersistentVolumeClaim) bool {
+	shouldProvision := isCorrectNode(pvc.GetAnnotations(), p.nodeName)
+	
+	if shouldProvision {
+		pvCapacity, err := calculatePvCapacity(p.pvDir)
+		if pvCapacity != nil && pvCapacity.Cmp(pvc.Spec.Resources.Requests[v1.ResourceName(v1.ResourceStorage)]) < 0 {
+			glog.Error("PVC request size larger than total possible PV size")
+			shouldProvision = false;
+		} else if pvCapacity == nil {
```

Comment:
Why not short circuit this with `if err != nil`

Reply:
Done